components/gateway/pkg/proxy/transport.go

```diff
@@ -62,13 +64,42 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 	resp.Body = ioutil.NopCloser(bytes.NewReader(responseBody))
 	defer httpcommon.CloseBody(resp.Body)
 
+	isMutation, err := checkQueryType(requestBody, "mutation")
+	if err != nil {
+		return nil, err
+	}
+	if !isMutation {
+		log.Println("Will not auditlog queries")
+		return resp, nil
+	}
+
 	err = t.auditlogSvc.Log(string(requestBody), string(responseBody), claims)
 	if err != nil {
 		return nil, errors.Wrap(err, "while sending to auditlog")
 	}
 	return resp, nil
 }
 
+func checkQueryType(requestBody []byte, typee string) (bool, error) {
```

Comment:
typo in `typee` ?

Reply:
it should be typee because type is a reserved keyword...